api/agent/slots.go

```diff
@@ -158,19 +128,92 @@ func (a *slotQueue) ejectSlot(s *slotToken) bool {
 
 func (a *slotQueue) destroySlotQueue() {
 	doSignal := false
+	purgePool := make([]*slotToken, 0)
```

Comment:
nit: `var purgePool []*slotToken` won't allocate, and `range` statement will handle the `nil` value fine (and of course, the assignment statement will also still work) -- this method is called infrequently, not a big deal, just leaving a note